yarn-project/types/src/abi/contract_artifact.ts

```diff
@@ -103,7 +103,7 @@ function generateFunctionArtifact(fn: NoirCompiledContractFunction): FunctionArt
 
   // If the function is not unconstrained, the first item is inputs or CallContext which we should omit
   let parameters = fn.abi.parameters.map(generateFunctionParameter);
-  if (functionType !== 'unconstrained') {
+  if (hasKernelFunctionInputs(parameters)) {
```

Comment:
Prevents the indiscriminate stripping of the first noir argument present when generating the abi ( it removes the context )